design/how-ironic-works.md

```diff
@@ -0,0 +1,356 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# how-ironic-works
+
+## Table of Contents
+
+<!--ts-->
+   * [how-ironic-works](#how-ironic-works)
+      * [Table of Contents](#table-of-contents)
+      * [How ironic controls hardware](#how-ironic-controls-hardware)
+      * [How ironic boots hardware](#how-ironic-boots-hardware)
+      * [How ironic writes an operating system image to baremetal?](#how-ironic-writes-an-operating-system-image-to-baremetal)
+      * [What connectivity is required?](#what-connectivity-is-required)
+      * [How-to](#how-to)
+         * [How to discover hardware?](#how-to-discover-hardware)
+         * [How to add hardware to ironic?](#how-to-add-hardware-to-ironic)
+      * [Updating information about a hardware node in ironic.](#updating-information-about-a-hardware-node-in-ironic)
+         * [How do I identify the current state?](how-do-i-identify-the-current-state)
+         * [How to inspect hardware?](#how-to-inspect-hardware)
+         * [How to deploy?](#how-to-deploy)
+         * [How to unprovision a baremetal node?](#how-to-unprovision-a-baremetal-node)
+         * [How to delete a baremetal node?](#how-to-delete-a-baremetal-node)
+         * [How to create the record of an active node](#how-to-create-the-record-of-an-active-node)
+      * [References](#references)
+<!--te-->
+
+This document explains how to use ironic in order to achieve various
+tasks such as creating a node, recreating a node, unprovisioning a
+node, deleting a node.
+
+This is not intended to be design specific documentation, but intends
+to help convey the mechanics of how, such that the reader does not
+have to become an expert in Ironic in order to learn to leverage it.
+
+## How ironic controls hardware
+
+Ironic is largely designed around the ability to issue commands to a
+remote Baseboard Management Controller (BMC) in order to assert the
+desired next boot device and control system power state.
+
+BMCs are also not required, however lacking BMC functionality, then
+power and potential boot mode changes need to be performed by an
+external entity such as a human or a network attached power distribution
+unit.
+
+- Access to the BMC can be via a routed IP network,
+  however this may be less desirable from a security standpoint.
+
+- Some vendor driver interfaces may expect to assert configuration
+  to a BMC where the BMC is expected to reach controller resources
+  or hosted services.
+  Mainly this should be expected when virtual media is in use, which means
+  some sort of dedicated BMC network access becomes desirable, as long as
+  the connectivity matches the failure domain of the machines upon which
+  ironic will be executing.
```

Comment:
I think what this is saying is that when we use virtual media we (or the deployer) have to ensure that the BMC can see the place where the media image is being hosted. Is that right?

You say this is the "main" case where the issue applies. What other cases do we need to consider?

Reply:
That is exactly what I'm trying to convey. I think the other issue might end up being lower end more general purpose processors trying to be an IO support mechanism with virtual media. I think at that point if we focus on reducing ramdisk size, we won't even notice that as a possible bottleneck.

Reply:
OK. We can iterate on the wording.